x-pack/plugins/security_solution/public/management/components/artifact_list_page/artifact_list_page.tsx

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useMemo, useState } from 'react';
+
+import type { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { EuiButton, EuiSpacer, EuiText } from '@elastic/eui';
+import { EuiFlyoutSize } from '@elastic/eui/src/components/flyout/flyout';
+import { useLocation } from 'react-router-dom';
+import { AdministrationListPage } from '../administration_list_page';
+
+import { PaginatedContent, PaginatedContentProps } from '../paginated_content';
+
+import { ArtifactEntryCard } from '../artifact_entry_card';
+
+import { ArtifactListPageLabels, artifactListPageLabels } from './translations';
+import { useTestIdGenerator } from '../hooks/use_test_id_generator';
+import { ManagementPageLoader } from '../management_page_loader';
+import { SearchExceptions } from '../search_exceptions';
+import {
+  useArtifactCardPropsProvider,
+  UseArtifactCardPropsProviderProps,
+} from './hooks/use_artifact_card_props_provider';
+import { NoDataEmptyState } from './components/no_data_empty_state';
+import { ArtifactFlyoutProps, MaybeArtifactFlyout } from './components/artifact_flyout';
+import { useIsFlyoutOpened } from './hooks/use_is_flyout_opened';
+import { useSetUrlParams } from './hooks/use_set_url_params';
+import { useWithArtifactListData } from './hooks/use_with_artifact_list_data';
+import { ExceptionsListApiClient } from '../../services/exceptions_list/exceptions_list_api_client';
+import { ArtifactListPageUrlParams } from './types';
+import { useUrlParams } from './hooks/use_url_params';
+import { ListPageRouteState, MaybeImmutable } from '../../../../common/endpoint/types';
+import { DEFAULT_EXCEPTION_LIST_ITEM_SEARCHABLE_FIELDS } from '../../../../common/endpoint/service/artifacts/constants';
+import { ArtifactDeleteModal } from './components/artifact_delete_modal';
+import { useGetEndpointSpecificPolicies } from '../../services/policies/hooks';
+import { getLoadPoliciesError } from '../../common/translations';
+import { useToasts } from '../../../common/lib/kibana';
+import { useMemoizedRouteState } from '../../common/hooks';
+import { BackToExternalAppSecondaryButton } from '../back_to_external_app_secondary_button';
+import { BackToExternalAppButton } from '../back_to_external_app_button';
+
+type ArtifactEntryCardType = typeof ArtifactEntryCard;
+
+type ArtifactListPagePaginatedContentComponent = PaginatedContentProps<
+  ExceptionListItemSchema,
+  ArtifactEntryCardType
+>;
+
+export interface ArtifactListPageProps {
+  apiClient: ExceptionsListApiClient;
+  /** The artifact Component that will be displayed in the Flyout for Create and Edit flows */
+  ArtifactFormComponent: ArtifactFlyoutProps['FormComponent'];
```

Comment:
Yeah, but I find it easier to read the code using this approach. Also, using your approach would require me import the actual flyout component just so that I can get its `typeof` which seems like more verbose and unnecessary

Reply:
Yeah, `React.ComponentProps` is best to use in places where we're importing and using the component as well.